### PR TITLE
Update README.md for notifications and builder methods.

### PR DESCRIPTION
**Pull request by mikecdavis**
## Summary
- Add docs for builder methods
- Add docs for update config notification

### CODE REVIEW

File: core-httpclient-impl/README.md [hunk 1]
```diff
@@ -139,20 +151,49 @@ The polling interval is used to specify a fixed delay between consecutive HTTP r
...
-|`http.project.config.manager.blocking.unit`|SECONDS|Time unit corresponding to blocking duration|
-|`http.project.config.manager.sdk.key`|null|Optimizely project SDK key|
+|**http.project.config.manager.polling.duration**|5|Fixed delay between fetches for the datafile|
+|**http.project.config.manager.polling.unit**|MINUTES|Time unit corresponding to polling interval|
+|**http.project.config.manager.blocking.duration**|10|Maximum time to wait for initial bootstrapping|
+|**http.project.config.manager.blocking.unit**|SECONDS|Time unit corresponding to blocking duration|
+|**http.project.config.manager.sdk.key**|null|Optimizely project SDK key|
 
+## Update Config Notifications
+A notifcation signal will be triggered whenever a _new_ datafile is fetched. To subscribe to these notifications you can
```
Comment (aliabbasrizvi) [inline core-httpclient-impl/README.md]: Typo in notification

File: core-httpclient-impl/README.md [hunk 2]
```diff
@@ -139,20 +151,49 @@ The polling interval is used to specify a fixed delay between consecutive HTTP r
 #### Initial datafile
 
 You can provide an initial datafile via the builder to bootstrap the `ProjectConfigManager` so that it can be used
-immediately without blocking execution.
+immediately without blocking execution. This also serves as a "fallback" datafile if no HTTP connection can be established.
+This can be useful in a mobile environment where internet connectivity is not always guaranteed. This datafile will be
+discarded, however, after the first successful datafile poll.
+
```
Comment (hillaryfraley-optimizely) [inline core-httpclient-impl/README.md]: Suggest revising to:

The initial datafile also serves as a fallback datafile if HTTP connection cannot be established. This is useful in mobile environments, where internet connectivity is not guaranteed. The initial datafile will be discarded after the first successful datafile poll.

File: core-httpclient-impl/README.md [hunk 3]
```diff
@@ -139,20 +151,49 @@ The polling interval is used to specify a fixed delay between consecutive HTTP r
...
+### Builder Methods
+The following builder methods can be used to custom configure the `HttpProjectConfigManager`.
+
+|Builder Method|Default Value|Description|
+|---|---|---|
+|`withDatafile(String)`|null|Initial datafile, typically sourced from a local cached source.|
+|`withUrl(String)`|null|Url override location used to specify custom HTTP source for the Optimizely datafile.|
```
Comment (hillaryfraley-optimizely) [inline core-httpclient-impl/README.md]: In description, change "Url" to "URL" maybe?

File: core-httpclient-impl/README.md [hunk 4]
```diff
@@ -139,20 +151,49 @@ The polling interval is used to specify a fixed delay between consecutive HTTP r
...
+|`withFormat(String)`|https://cdn.optimizely.com/datafiles/%s.json|Parameterized datafile URL by SDK key.|
+|`withPollingInterval(Long, TimeUnit)`|5 minutes|Fixed delay between fetches for the datafile.|
```
Comment (hillaryfraley-optimizely) [inline core-httpclient-impl/README.md]: Does the value actually include "minutes"? Or is it more like the default value is 5 and the description should say "In minutes."

File: core-httpclient-impl/README.md [hunk 5]
```diff
@@ -139,20 +151,49 @@ The polling interval is used to specify a fixed delay between consecutive HTTP r
...
+|`withBlockingTimeout(Long, TimeUnit)`|10 seconds|Maximum time to wait for initial bootstrapping.|
```
Comment (hillaryfraley-optimizely) [inline core-httpclient-impl/README.md]: Same question as for previous method. Should default value be 10 and description include "In seconds."

File: core-httpclient-impl/README.md (hunk 4 above)
Comment (mikecdavis) [reply core-httpclient-impl/README.md]: The method takes two values, the "period" and the "time unit" so I think saying "5 minutes" is appropriate.

File: core-httpclient-impl/README.md (hunk 5 above)
Comment (mikecdavis) [reply core-httpclient-impl/README.md]: The method takes two values, the "period" and the "time unit" so I think saying "10 seconds" is appropriate.

**Pull request merged**